packages/@uppy/dashboard/src/components/FileCard/index.js

```diff
@@ -86,6 +86,11 @@ class FileCard extends Component {
   render () {
     const file = this.props.files[this.props.fileCardFor]
 
+    let showEditButton
+    this.props.editors.forEach((target) => {
+      showEditButton = this.props.getPlugin(target.id).сanEditFile(file)
+    })
```

Comment:
I think the looping stuff is probably the only way, but it could use `.some()` instead of `.forEach()`, so it will also stop looping as soon as it finds a match:
```suggestion
    const showEditButton = this.props.editors.some((target) => {
      const plugin = this.props.getPlugin(target.id)
      plugin.сanEditFile(file)
    })
```

(Also this is a minor thing, but I would probably put this entire code in the Dashboard plugin class itself and pass it in as a prop, so you can do `this.props.canEditFile(file)` instead of writing the loop in the render code. Then it doesn't depend directly on other plugins. If we ever end up doing testing for the components it will be easier that way)

Reply:
Great suggestion, done, thanks!